File: internal/templates/data/common/js/anthropic.ts
import type { AgentRequest, AgentResponse, AgentContext } from "@agentuity/sdk";
import Anthropic from '@anthropic-ai/sdk';

const anthropic = new Anthropic();

export default async function Agent(
	req: AgentRequest,
	resp: AgentResponse,
	ctx: AgentContext,
) {
	const message = await anthropic.messages.create({
    model: 'claude-3-opus-20240229',
    max_tokens: 1024,
    messages: [{ role: 'user', content: req.text() ??'Hello, Claude' }],
  });
	return resp.text(message.content[0].text);
}

File: internal/templates/data/common/py/anthropic.py
from agentuity.server.types import AgentRequest, AgentResponse, AgentContext
from anthropic import Anthropic

client = Anthropic()


def run(request: AgentRequest, response: AgentResponse, context: AgentContext):
    result = client.messages.create(
        max_tokens=1024,
        messages=[
            {
                "role": "user",
                "content": request.text or "Hello, Claude",
            }
        ],
        model="claude-3-5-sonnet-latest",
    )
    return response.text(result.content[0].text)
